### DISCUSSION TITLE
meow-keypad and reverse-im

### DISCUSSION THREAD

**Discussion by ghost** [meow-edit/meow#211] [Q&A] [answered]
Has anyone been able to combine meow-keypad with [reverse-im](https://github.com/a13/reverse-im.el)? Normal mode seems to work fine, but meow-keypad just inserts the symbol without translation.

**Comment by DogLooksGood** [comment]
I have no experience with reverse-im. But it looks like you can toggle reverse-im when enter and exit keypad state?
There's a `meow-keypad-mode-hook`.

**Reply by ghost** [reply]
I'm still not that great with elisp, so maybe I'm wrong, but as far as I can see reverse-im either [modifies](https://github.com/a13/reverse-im.el/blob/50b8376f152916bc200635a112db9439bc3cc9b5/reverse-im.el#L199) keymap with `set-keymap-parent` or [advises](https://github.com/a13/reverse-im.el/blob/50b8376f152916bc200635a112db9439bc3cc9b5/reverse-im.el#L299) functions that create transient keymap, which is [problematic](https://github.com/a13/reverse-im.el/issues/17#issuecomment-604018818). I haven't looked closely on how meow-keypad works, but advising it doesn't help, so maybe there's a way to make a hack for meow-keypad to use [this](https://github.com/a13/reverse-im.el/blob/50b8376f152916bc200635a112db9439bc3cc9b5/reverse-im.el#L311) function directly?

**Reply by ghost** [reply]
It overrides function-key-map specifically.

**Comment by ghost** [comment, marked as answer]
I hacked it together
``` elisp
(defun +translate-input-event (event)
  "Use reverse-im to translate input EVENT."
  (if (numberp event)
      (reverse-im--translate-char event t)
    event))

(advice-add 'meow--event-key :filter-return #'+translate-input-event)
```
